File: eir-back/src/main/kotlin/com/eir/gdr/entities/Perk.kt
package com.eir.gdr.entities

import io.vertx.ext.sql.ResultSet

data class Perk (
    var id: Int? = null,
    var name: String? = null,
    var description: String? = null,
    var negative: Boolean? = null,
    var affectedCharacteristic: List<Effect>? = null
) {
    companion object {
        fun getPerks(rs: ResultSet): List<Perk> =
            rs.rows.map { r ->
                Perk(
                    r.getInteger("id"),
                    r.getString("name"),
                    r.getString("description"),
                    r.getInteger("isNegative") == 1
                )
            }
    }
}

File: eir-front-angular/src/app/services/dtos/perk.ts
import {Characteristic} from './characteristic';

export interface PerkCharacteristic {
  characteristic: Characteristic;
  value: number;
}

export interface Perk {
  id: number;
  name: string;
  description: string;
  negative: boolean;
  affectedCharacteristic: PerkCharacteristic[];
}
